### PR TITLE
feat: dynamic slack messages

### LINKED ISSUE #1511: [🔨 Enhancement]: Slack provider should support editing messages
For example:
1. Alert arrives firing
2. Workflow runs and send a slack message
3. Alert arrives as resolved
4. Workflow runs and edits the respective message to reflect the status change

This can be done utilizing the alert enrichment (`enrich_alert`) mechanism.

### PR DESCRIPTION
**Pull request by 35C4n0r**
Closes  #1511
/claim  #1511

## ✅ Checks

- [x] My pull request adheres to the code style of this project
- [x] My code requires changes to the documentation
- [x] I have updated the documentation as required
- [x] All the tests have passed

## ℹ Additional Information

https://www.loom.com/share/b9a56255f9bb4e2e8251d24a16a12029?sid=28af7acf-067f-4895-bf7b-0794b46d8407

### CODE REVIEW

File: keep/providers/slack_provider/slack_provider.py [hunk 1]
```diff
@@ -103,14 +103,17 @@ def oauth2_logic(**payload) -> dict:
...
         """
         Notify alert message to Slack using the Slack Incoming Webhook API
         https://api.slack.com/messaging/webhooks
 
         Args:
             kwargs (dict): The providers with context
         """
+        notify_data = None
         self.logger.info(
             f"Notifying message to Slack using {'webhook' if self.authentication_config.webhook_url else 'access token'}",
```
Comment (talboren) [inline keep/providers/slack_provider/slack_provider.py:118]: Is it not possible to edit a message using webhooks? Can't you send ts in here as well?

**Review by talboren — COMMENTED** [review]
1 small commment but lgtm

File: keep/providers/slack_provider/slack_provider.py (hunk 1 above)
Comment (35C4n0r) [reply keep/providers/slack_provider/slack_provider.py:118]: @talboren posting to the webhook only returns the 'ok' as the response. Timestamp is necessary to edit a message.

Comment (35C4n0r) [reply keep/providers/slack_provider/slack_provider.py:118]: There's no direct way but we can try

**Review by talboren — APPROVED** [review]
lgtm

**Pull request merged**